src/simplnx/Utilities/ThreadedAlgorithm.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include "simplnx/Filter/IFilter.hpp"
+#include "simplnx/simplnx_export.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+namespace nx::core
+{
+
+/**
+ * @brief
+ */
+class SIMPLNX_EXPORT FilterAlgorithm
+{
+public:
+  FilterAlgorithm(const IFilter::MessageHandler& messageHandler, const std::atomic_bool& shouldCancel);
+
+  virtual ~FilterAlgorithm() noexcept = default;
```

Comment:
We don't actually have to mark the destructors as `virtual` because these classes aren't intended to be used polymorphically. Not a big deal either way though.

Reply:
The subclasses implement the destructor, even though they are "default" implementations, so core guidelines say that you should mark them override. And when you mark them override, then it complains that the base class isn't marked virtual. Thoughts?

Reply:
Do you have the specific guideline that comes up?

Reply:
Is it just this one or are there two?